src/app/Topology/Shared/Entity/EntityDetails.tsx

```diff
@@ -0,0 +1,598 @@
+                    <TargetResourceItem key={val} targetNode={targetNode} resourceType={val} />
+                  ))}
+                </Tbody>
+              </TableComposable>
+            </CardBody>
+          </Card>
+        </StackItem>
+      ))}
+    </Stack>
+  );
+};
+
+export const TargetResourceItem: React.FC<{
+  targetNode: TargetNode;
+  resourceType: TargetOwnedResourceType | TargetRelatedResourceType;
+}> = ({ targetNode, resourceType, ...props }) => {
+  const services = React.useContext(ServiceContext);
+  const addSubscription = useSubscriptions();
+  const targetSubjectRef = React.useRef(new Subject<TargetNode>());
+  const targetSubject = targetSubjectRef.current;
+
+  const [resources, setResources] = React.useState<ResourceTypes[]>([]);
+  const [loading, setLoading] = React.useState(true);
+  const [error, setError] = React.useState<Error>();
+
+  const isOwned = React.useMemo(() => isOwnedResource(resourceType), [resourceType]);
+
+  React.useEffect(() => {
+    addSubscription(
+      targetSubject.pipe(switchMap((tn) => getTargetOwnedResources(resourceType, tn, services.api))).subscribe({
+        next: (rs) => {
+          setLoading(false);
+          setError(undefined);
+          setResources(rs);
+        },
+        error: (error) => {
+          setLoading(false);
+          setError(error);
+        },
+      })
+    );
+  }, [setLoading, addSubscription, setResources, resourceType, services.api, targetSubject]);
+
+  React.useEffect(() => {
+    addSubscription(
+      targetSubject
+        .pipe(
+          switchMap((tn) =>
+            combineLatest([
+              of(tn),
+              merge(...getResourceAddedEvents(resourceType).map((cat) => services.notificationChannel.messages(cat))),
+            ])
+          )
+        )
+        .subscribe(([targetNode, event]) => {
+          const extractedUrl = extraTargetConnectUrlFromEvent(event);
+          if (!isOwned || (extractedUrl && extractedUrl === targetNode.target.connectUrl)) {
+            setLoading(true);
+            setResources((old) => {
+              // Avoid accessing state directly, which
+              // causes the effect to run every time
+              addSubscription(
+                getResourceListPatchFn(resourceType, targetNode, services.api)(old, event, false).subscribe({
+                  next: (rs) => {
+                    setLoading(false);
+                    setError(undefined);
+                    setResources(rs);
+                  },
+                  error: (error) => {
+                    setLoading(false);
+                    setError(error);
+                  },
+                })
+              );
+              return old;
+            });
+          }
+        })
+    );
+  }, [
+    addSubscription,
+    setLoading,
+    services.api,
+    isOwned,
+    targetSubject,
+    resourceType,
+    services.notificationChannel,
+    setResources,
+    setError,
+  ]);
+
+  React.useEffect(() => {
```

Comment:
Are there duplicate useEffects here or am I missing something?...

If there is not, there seems to be a lot of duplicated code, so I think there could be some refactoring here...

Reply:
Nope this one is for DELETE events. The one above is for ADD event.

Reply:
But I think there might be a way to refactor it.

Reply:
Refactored :D